pkg/controller/certificates/trigger/policies/gatherer_test.go

```diff
@@ -0,0 +1,189 @@
+				gen.SetCertificateUID("uid-7"),
+				gen.SetCertificateSecretName("secret-1"),
+				gen.SetCertificateRevision(7),
+			),
+			wantRequest: gen.CertificateRequest("cr-7", gen.SetCertificateRequestNamespace("default-unit-test-ns"),
+				gen.AddCertificateRequestOwnerReferences(gen.CertificateRef("cert-1", "uid-7")),
+				gen.AddCertificateRequestAnnotations(map[string]string{
+					"cert-manager.io/certificate-revision": "7",
+				}),
+			),
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			fakeClockStart, _ := time.Parse(time.RFC3339, "2021-01-02T15:04:05Z07:00")
+			log := logtest.TestLogger{T: t}
+			redirectKlog(t, log)
+
+			test.builder.T = t
+			test.builder.Clock = fakeclock.NewFakeClock(fakeClockStart)
+
+			// In this test, we do not use Register(controller.Context).
+			// The Register(controller.Context) usually takes care of
+			// triggering the init() func in ./pkg/api/scheme.go. If we
+			// forget to have the init() func called, the apiVersion and
+			// kind fields on cert-manager objects are not automatically
+			// filled, which breaks the lister cache (i.e., the "indexer").
+			_ = cmscheme.Scheme
+
+			test.builder.Init()
+
+			// One weird behavior in client-go is that listers won't return
+			// anything if no event handler has been registered on this
+			// type's informer. This is because the "indexers" (i.e., the
+			// client-go cache) being lazily created. In our case, the
+			// indexer for the CR type only gets created if we register an
+			// event handler on the CR informer. And since we do not use
+			// the Register(controller.Context) in these lister-only unit
+			// tests, we "force" the creation of the indexer for the CR
+			// type by registering a fake handler.
+			test.builder.SharedInformerFactory.Certmanager().V1().CertificateRequests().Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+				AddFunc: func(obj interface{}) {},
+			})
+
+			// Even though we are only relying on listers in this unit test
+			// and do not use the informer event handlers, we still need to
+			// start the informers since the listers would return nothing
+			// otherwise (see above comment).
+			test.builder.Start()
+
+			defer test.builder.CheckAndFinish()
+
+			// NOTE(mael): I wanted to make sure that (1) the lister was
+			// called using the right namespace as its input argument, and
+			// (2) the lister was actually called (or not called).
+			//
+			// The reason for (1) comes from the fact that when the lister
+			// returns an empty list, the empty list might be due to two
+			// different causes: either the lister is called with an
+			// unexpected namespace instead of the non-empty expected
+			// namespace, or the lister is called with the right namespace
+			// and the fake clientset behaved as expected. In order to
+			// avoid the inconsistancy, we do make sure to have the right
```

Comment:
```suggestion
			// avoid the inconsistency, we do make sure to have the right
```
(I have an action that could catch typos like this 😉)

Reply:
I took a look at your Github Action, it looks neat!